File: module-7-evals/solutions/run-evals.ts
/**
 * Module 7: Run evaluations
 * npx tsx module-7-evals/run-evals.ts
 */
import "dotenv/config";
import { runEvals } from "./harness.js";
import { computeTaskResult, computeReport } from "./metrics.js";
import { printReport } from "./report.js";
import { codingTasks } from "./tasks/coding-tasks.js";

async function main() {
  const trialsPerTask = Number(process.argv[2]) || 3;
  console.log(`Running ${codingTasks.length} tasks × ${trialsPerTask} trials each\n`);

  const trials = await runEvals(codingTasks, trialsPerTask);

  // Compute results per task
  const taskResults = codingTasks.map((task) => {
    const taskTrials = trials.filter((t) => t.taskId === task.id);
    return computeTaskResult(task.id, taskTrials);
  });

  const report = computeReport(taskResults);
  printReport(report);
}

main().catch(console.error);

File: module-7-evals/solutions/python/run_evals.py
"""Module 7: Run evaluations."""
import sys
import os

sys.path.insert(0, os.path.dirname(__file__))

from dotenv import load_dotenv
load_dotenv()

from harness import run_evals
from metrics import compute_task_result, compute_report
from report import print_report
from tasks.coding_tasks import coding_tasks


def main():
    trials_per_task = int(sys.argv[1]) if len(sys.argv) > 1 else 3
    print(f"Running {len(coding_tasks)} tasks × {trials_per_task} trials each\n")

    trials = run_evals(coding_tasks, trials_per_task)
    task_results = [
        compute_task_result(task.id, [t for t in trials if t.task_id == task.id])
        for task in coding_tasks
    ]
    report = compute_report(task_results)
    print_report(report)


if __name__ == "__main__":
    main()
